http/src/main/resources/jupyter/gce-init.sh

```diff
@@ -0,0 +1,528 @@
+    # `docker-compose up` is not retried.
+    COMPOSE_FILES=(-f /etc/`basename ${PROXY_DOCKER_COMPOSE}`)
+    cat /etc/`basename ${PROXY_DOCKER_COMPOSE}`
+    if [ ! -z ${JUPYTER_DOCKER_IMAGE} ] ; then
+      COMPOSE_FILES+=(-f /etc/`basename ${JUPYTER_DOCKER_COMPOSE}`)
+      cat /etc/`basename ${JUPYTER_DOCKER_COMPOSE}`
+    fi
+    if [ ! -z ${RSTUDIO_DOCKER_IMAGE} ] ; then
+      COMPOSE_FILES+=(-f /etc/`basename ${RSTUDIO_DOCKER_COMPOSE}`)
+      cat /etc/`basename ${RSTUDIO_DOCKER_COMPOSE}`
+    fi
+    if [ ! -z ${WELDER_DOCKER_IMAGE} ] && [ "${WELDER_ENABLED}" == "true" ] ; then
+      COMPOSE_FILES+=(-f /etc/`basename ${WELDER_DOCKER_COMPOSE}`)
+      cat /etc/`basename ${WELDER_DOCKER_COMPOSE}`
+    fi
+
+    retry 5 docker-compose "${COMPOSE_FILES[@]}" config
+    retry 5 docker-compose "${COMPOSE_FILES[@]}" pull
+    retry 5 docker-compose "${COMPOSE_FILES[@]}" up -d
+
+    STEP_TIMINGS+=($(date +%s))
+
+    # If we have a service account JSON file, create an .env file to set GOOGLE_APPLICATION_CREDENTIALS
+    # in the docker container. Otherwise, we should _not_ set this environment variable so it uses the
+    # credentials on the metadata server.
+    if [ ! -z ${SERVICE_ACCOUNT_CREDENTIALS} ] ; then
+      if [ ! -z ${JUPYTER_DOCKER_IMAGE} ] ; then
+        log 'Copying SA into Jupyter Docker...'
+        docker cp /etc/${SERVICE_ACCOUNT_CREDENTIALS} ${JUPYTER_SERVER_NAME}:/etc/${SERVICE_ACCOUNT_CREDENTIALS}
+      fi
+      if [ ! -z ${RSTUDIO_DOCKER_IMAGE} ] ; then
+        log 'Copying SA into RStudio Docker...'
+        docker cp /etc/${SERVICE_ACCOUNT_CREDENTIALS} ${RSTUDIO_SERVER_NAME}:/etc/${SERVICE_ACCOUNT_CREDENTIALS}
+      fi
+      if [ ! -z ${WELDER_DOCKER_IMAGE} ] && [ "${WELDER_ENABLED}" == "true" ] ; then
+        log 'Copying SA into Welder Docker...'
+        docker cp /etc/${SERVICE_ACCOUNT_CREDENTIALS} ${WELDER_SERVER_NAME}:/etc/${SERVICE_ACCOUNT_CREDENTIALS}
+      fi
+    fi
+
+    # if Welder is installed, start the service.
+    # See https://broadworkbench.atlassian.net/browse/IA-1026
+    if [ ! -z ${WELDER_DOCKER_IMAGE} ] && [ "${WELDER_ENABLED}" == "true" ] ; then
+      log 'Starting Welder file synchronization service...'
+      retry 3 docker exec -d ${WELDER_SERVER_NAME} /opt/docker/bin/entrypoint.sh
+    fi
+
+    STEP_TIMINGS+=($(date +%s))
+
+    # Jupyter-specific setup, only do if Jupyter is installed
+    if [ ! -z ${JUPYTER_DOCKER_IMAGE} ] ; then
+      log 'Installing Jupydocker kernelspecs...'
+
+      # Change Python and PySpark 2 and 3 kernel specs to allow each to have its own spark
+      # TODO This is baked into terra-jupyter-base as of version 0.0.6. Keeping it here for now to support prior image versions.
+      retry 3 docker exec -u root ${JUPYTER_SERVER_NAME} ${JUPYTER_SCRIPTS}/kernel/kernelspec.sh ${JUPYTER_SCRIPTS}/kernel ${KERNELSPEC_HOME}
+
+      # Install hail addition if the image is old leonardo jupyter image or it's a hail specific image
```

Comment:
Can probably remove this, we won't support hail on GCE

Reply:
👍